persistence/test/benchmark_state_test.go

```diff
@@ -0,0 +1,174 @@
+package test
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	"reflect"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/pokt-network/pocket/persistence"
+	"github.com/pokt-network/pocket/persistence/indexer"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+const (
+	maxStringAmount = 1000000000000000000
+)
+
+var isModifierRe = regexp.MustCompile(`^(Insert|Set|Add|Subtract)`) // Add Update?
+
+// INVESTIGATE: This benchmark can be used to experiment with different Merkle Tree implementations
+// and key-value stores.
+// IMPROVE(#361): Improve the output of this benchmark to be more informative and human readable.
+func BenchmarkStateHash(b *testing.B) {
+	log.SetOutput(ioutil.Discard)
+	defer log.SetOutput(os.Stderr)
+
+	clearAllState()
+	b.Cleanup(clearAllState)
+
+	// NOTE: The idiomatic way to run Go benchmarks is to use `b.N` and the `-benchtime` flag,
+	// to specify how long the benchmark should take. However, the code below is non-idiomatic
+	// since our goal is to test a specific we use a fixed number of iterations
+	testCases := []struct {
+		numHeights     int64
+		numTxPerHeight int
+		numOpsPerTx    int
+	}{
+		{1, 1, 1},
+		{1, 1, 10},
+		{1, 10, 10},
+
+		{10, 1, 1},
+		{10, 1, 10},
+		{10, 10, 10},
+
+		// This takes a VERY long time to run
+		// {100, 1, 1},
+		// {100, 1, 100},
+		// {100, 100, 100},
+	}
+
+	for _, testCase := range testCases {
```

Comment:
no-op: nice, I missed some table-based tests :)

Reply:
I'm personally quite proud of the test harness I built for this feature :)

Reply:
you should! However....... tests are failing I believe 🧐